File: medium/ts/2390.ts
// https://leetcode.com/problems/removing-stars-from-a-string/?envType=study-plan-v2&envId=leetcode-75
// Time: O(n)
// Space: O(n)
function removeStars(s: string): string {
  let charStack: string[] = [];

  for (let i = 0; i < s.length; i++) {
    if (s[i] === "*") {
      charStack.pop();
    } else {
      charStack.push(s[i]);
    }
  }
  return charStack.join("");
}

File: medium/python/2390.py
# https://leetcode.com/problems/removing-stars-from-a-string/?envType=study-plan-v2&envId=leetcode-75
# Time and Space: O(n)
class Solution:
    def removeStars(self, s: str) -> str:
        char_stack = []

        for ele in s:
            if ele == "*":
                char_stack.pop()
            else:
                char_stack.append(ele)
        return "".join(char_stack)
